cli/internal/cmd/upgradeexecute.go

```diff
@@ -68,18 +70,32 @@ func upgradeExecute(cmd *cobra.Command, imageFetcher imageFetcher, upgrader clou
 		return displayConfigValidationErrors(cmd.ErrOrStderr(), err)
 	}
 
-	helm, err := cmd.Flags().GetBool("helm")
+	helmFlag, err := cmd.Flags().GetBool("helm")
 	if err != nil {
 		return err
 	}
-	if helm {
+	if helmFlag {
 		timeout, err := cmd.Flags().GetDuration("timeout")
 		if err != nil {
 			return err
 		}
-		if err := upgrader.UpgradeHelmServices(cmd.Context(), conf, timeout); err != nil {
+		err = upgrader.UpgradeHelmServices(cmd.Context(), conf, timeout, false)
+		if errors.Is(err, helm.ErrConfirmationMissing) {
+			cmd.PrintErrln("WARNING: Upgrading cert-manager will destroy all custom resources you have manually created that are based on the current version of cert-manager.")
+			ok, askErr := askToConfirm(cmd, "Do you want to upgrade cert-manager anyways?")
+			if askErr != nil {
+				return fmt.Errorf("asking for confirmation: %w", err)
+			}
+			if !ok {
+				cmd.Println("Aborting upgrade.")
+				return nil
+			}
```

Comment:
This should only happen if `yes` flag isn't set, compare create command.

Reply:
As this will require parsing another flag, I'd also propose to add a parseUpgradeExecFlags function in the same style as many other commands have.